msticpy/nbtools/azure_ml_tools.py

```diff
@@ -0,0 +1,425 @@
+    sys_ver = _get_pkg_version(sys.version_info[:3])
+    _disp_html("Checking Python kernel version...")
+    if sys_ver < min_py_ver:
+        _disp_html(
+            f"""
+            <h4><font color='red'>This notebook requires a later
+            (Python) kernel version.</h4></font>
+            Select a kernel from the notebook toolbar (above), that is Python
+            {min_py_ver} or later (Python 3.8 recommended)<br>
+            """
+        )
+        _disp_html(
+            f"""
+            Please see the <a href="{TROUBLE_SHOOTING}">TroubleShootingNotebooks</a>
+            for more information<br><br><hr>
+            """
+        )
+        raise RuntimeError(f"Python {min_py_ver} or later kernel is required.")
+
+    if sys_ver < _get_pkg_version("3.8"):
+        _disp_html(
+            "Recommended: switch to using the 'Python 3.8 - AzureML' notebook kernel"
+            " if this is available."
+        )
+    _disp_html(f"Info: Python kernel version {sys_ver} OK<br>")
+
+
+def _check_mp_install(min_mp_ver, mp_release, extras):
+    """Check for and try to install required MSTICPy version."""
+    # Use the release ver specified in params, in the environment or
+    # the notebook default.
+    pkg_version = _get_pkg_version(min_mp_ver)
+    mp_install_version = mp_release or os.environ.get("MP_TEST_VER", str(pkg_version))
+
+    check_mp_ver(min_msticpy_ver=mp_install_version, extras=extras)
+
+
+def check_mp_ver(min_msticpy_ver, extras):
+    """
+    Check and optionally update the current version of msticpy.
+
+    Parameters
+    ----------
+    min_msticpy_ver : Tuple[int, int]
+        Minimum MSTICPy version
+    extras : Optional[List[str]], optional
+        A list of extras required for MSTICPy
+
+    Raises
+    ------
+    ImportError
+        If MSTICPy version is insufficient and we need to upgrade
+
+    """
+    mp_min_pkg_ver = _get_pkg_version(min_msticpy_ver)
+
+    _disp_html("Checking msticpy version...<br>")
+
+    inst_version = _get_pkg_version(__version__)
+    if inst_version < mp_min_pkg_ver:
+        _disp_html(
+            MISSING_PKG_ERR.format(
+                package="msticpy",
+                inst_ver=inst_version,
+                req_ver=mp_min_pkg_ver,
+            )
+        )
+        mp_pkg_spec = f"msticpy[{','.join(extras)}]" if extras else "msticpy"
+        mp_pkg_spec = f"{mp_pkg_spec}>={min_msticpy_ver}"
+
+        _disp_html(
+            f"Please run the following command to upgrade MSTICPy<br>"
+            f"<pre>!{mp_pkg_spec}</pre><br>"
+        )
+        raise ImportError(
+            "Unsupported version of MSTICPy installed",
+            f"Installed version: {inst_version}",
+            f"Required version: {mp_min_pkg_ver}",
+        )
+
+    _disp_html(f"Info: msticpy version {mp_min_pkg_ver} OK<br>")
+
+
+def _set_kql_env_vars(extras):
+    jp_extended = ("azsentinel", "azuresentinel", "kql")
+    # If running in
```

Comment:
Appears to be a truncated comment

Reply:
Yep. Couldn't remember what I was going to write there.
However, I did add docstrings to all the funcs and type hints.